### PR TITLE
three controllers and their dtos/mappers were implemented

### PR DESCRIPTION
**Pull request by oleksiistepaniak**

### CODE REVIEW

**Review by yurihhhT — APPROVED** [review]
good job, mate!

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailable(@RequestParam Long movieId, @RequestParam
+            @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date).stream()
+                .map(movieSessionMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("{/id}")
```
Comment (nacenik) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:49]: You should put a forward slash before the path variable 
```suggestion
    @PutMapping("/{id}")
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public MovieSessionResponseDto update(@PathVariable Long id, @RequestBody
+            MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionMapper.toModel(movieSessionRequestDto);
+        movieSession.setId(id);
+        return movieSessionMapper.toDto(movieSessionService.update(movieSession));
+    }
+
+    @DeleteMapping("{/id}")
```
Comment (nacenik) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:57]: You should put a forward slash before the path variable 
```suggestion
    @DeleteMapping("/{id}")
```

File: src/main/java/mate/academy/spring/dao/MovieSessionDao.java [hunk 3]
```diff
@@ -9,4 +9,8 @@ public interface MovieSessionDao extends GenericDao<MovieSession> {
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     Optional<MovieSession> get(Long id);
+
+    MovieSession update(MovieSession movieSession);
+
+    void remove(Long id);
```
Comment (nacenik) [inline src/main/java/mate/academy/spring/dao/MovieSessionDao.java:15]: better make it boolean for better testing possibilities

File: src/main/java/mate/academy/spring/service/MovieSessionService.java [hunk 4]
```diff
@@ -10,4 +10,8 @@ public interface MovieSessionService {
     MovieSession get(Long id);
 
     MovieSession add(MovieSession session);
+
+    MovieSession update(MovieSession movieSession);
+
+    void remove(Long id);
```
Comment (nacenik) [inline src/main/java/mate/academy/spring/service/MovieSessionService.java:16]: better make it boolean for better testing possibilities

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution! Let’s improve it 😉

**Review by petro-ushchuk — APPROVED** [review]
Good work 🔥

**Pull request closed without merging**